src/main/kotlin/pt/up/fe/ni/website/backend/service/PostService.kt

```diff
@@ -0,0 +1,36 @@
+package pt.up.fe.ni.website.backend.service
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import pt.up.fe.ni.website.backend.model.Post
+import pt.up.fe.ni.website.backend.repository.PostRepository
+
+class PostDto {
+    var title: String? = null
+    var body: String? = null
+    var thumbnailPath: String? = null
+}
+
+@Service
+class PostService(private val repository: PostRepository) {
+    fun getAllPosts(): List<Post> = repository.findAll().toList()
+
+    fun getPost(PostId: Long): Post =
+        repository.findByIdOrNull(PostId.toLong()) ?: throw NoSuchElementException("Post Not Found")
+
+    fun createPost(post: Post) = repository.save(post)
+
+    fun updatePost(PostId: Long, post: PostDto) {
+        val targetPost = repository.findByIdOrNull(PostId) ?: throw NoSuchElementException("Post Not Found")
+        targetPost.title = post.title ?: targetPost.title
+        targetPost.body = post.body ?: targetPost.body
+        targetPost.thumbnailPath = post.thumbnailPath ?: targetPost.thumbnailPath
+
+        repository.save(targetPost)
+    }
```

Comment:
I think that, for consistency, we should return an empty object (`{}`) instead of an empty body. Changing the function return type to `Map<String, String>` and returning a `mapOf()` seems to do the trick

Reply:
I committed the change where it returns the edited Post, I don't really understand what to do.